docs/api/events.md

```diff
@@ -0,0 +1,288 @@
+# KNoT Fog Connector Events API - v2.0.0
+
+This document describes the events `knot-fog-connector` is able to receive and send. They are gruped based on the external clients point of view, i.e. publishing or subscribing to the topics. In each section, it is provided information about the header, payload and protocol binding details of the event.
+
+## Content
+
+- [Publish](#publish) (external clients can publish to):
+  - [device.registered](#device-registered)
+  - [device.unregistered](#device-unregistered)
+  - [device.schema.updated](#device-schema-updated)
+  - [data.published](#data-published)
+
+- [Subscribe](#subscribe) (external clients can subscribe to):
+  - [data.request](#data-request)
+  - [data.update](#data-update)
+----------------------------------------------------------------
+
+## Publish
+
+This section describes the events that this service can receive from the external applications.
+
+### **device.registered** <a name="device-registered"></a>
+
+Event that represents a thing was registered.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  - `id` **String** thing's ID
+  - `token` **String** thing's token
+  - `error` **String** described the occurred error
+
+  Success example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "token": "5b67ce6bef21701331152d6297e1bd2b22f91787",
+    "error": null
+  }
+  ```
+
+  Error example:
+
+  ```json
+  {
+    "id": "3aa21010cda96fe9",
+    "token": "",
+    "error": "Device already exists"
+  }
+  ```
+
+</details>
+
+<details>
+  <summary>AMQP Binding</summary>
+
+  - Exchange:
+    - Type: direct
+    - Name: device
+    - Durable: `true`
+    - Auto-delete: `false`
+  - Routing key: device.registered
+
+</details>
+
+### **device.unregistered** <a name="device-unregistered"></a>
+
+Event that represents a thing was removed.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  - `id` **String** thing's ID
+  - `error` **String** described the occurred error
+
+  Success example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "error": null
+  }
+  ```
+
+  Error example:
+
+  ```json
+  {
+    "id": "3aa21010cda96fe9",
+    "error": "forbidden",
+  }
+  ```
+</details>
+
+<details>
+  <summary>AMQP Binding</summary>
+
+  - Exchange:
+    - Type: direct
+    - Name: device
+    - Durable: `true`
+    - Auto-delete: `false`
+  - Routing key: device.unregistered
+
+</details>
+
+### **device.schema.updated** <a name="device-schema-updated"></a>
+
+Event that represents a thing's schema was updated.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  - `id` **String** thing's ID
+  - `error` **String** a string with detailed error message
```

Comment:
We should standardize the description of `error`. In other definitions, we used `describes the occurred error` and this one is `a string with detailed error message`. The chosen description should be used on the other error descriptions throughout the documentation.

```suggestion
  - `error` **String** detailed error message
```

Reply:
Done.